scripts/gcovr

```diff
@@ -2097,7 +2097,25 @@ def print_xml_report(covdata):
 #
 # Create option parser
 #
-parser = OptionParser()
+
+# helper for percentage actions
+def check_percentage(option, opt, value):
+    try:
+        x = float(value)
+        if x < 0.0 or x > 100.0:
```

Comment:
I'd typically write `if not (0.0 <= x <= 100.0)` so that:

 - the condition we are validating for is spelled positively, and
 - we use chained comparisons to verify the range.

But this is mostly a personal preference. What do you think?

Reply:
It's much nicer. I'm still new to python so learning new things every time.